hazelcast/src/main/java/com/hazelcast/spi/impl/NodeEngineImpl.java

```diff
@@ -425,11 +425,11 @@ public void reset() {
 
     public void shutdown(final boolean terminate) {
         logger.finest("Shutting down services...");
+        operationService.shutdown();
```

Comment:
it calls `responseHandler.sendResponse()`. 
The remote response handler impl uses operation service. Currently it's not a problem because the response is sent only for [local invocations](https://github.com/jerrinot/hazelcast/blob/891660f3874acfcfe8cd7c56c73fb7c7ec36e8dd/hazelcast/src/main/java/com/hazelcast/spi/impl/waitnotifyservice/impl/WaitNotifyServiceImpl.java#L267-L267). but it still feels a bit fragile. 

that being said I don't know any better:)